apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.V1/PgNumeric.cs

```diff
@@ -0,0 +1,487 @@
+
+        /// <summary>
+        /// The maximum scale of numeric type.
+        /// </summary>
+        private const int MaxScale = 16383;
+
+        /// <summary>
+        /// The maximum precision of numeric type.
+        /// </summary>
+        private const int MaxPrecision = MaxDigitsBeforeDecimalPoint + MaxScale + 1;
+
+        /// <summary>
+        /// The string constant for not a number (NaN).
+        /// </summary>
+        private const string NotANumber = "NaN";
+
+        /// For a numeric(precision, scale) maximum and minimum values are defined as +- ((10^precision)-1)/(10^scale), so maximum possible numeric would be numeric
+        /// with max precision and zero scale, i.e. numeric(147455, 0), and its max and min values would be ((BigInteger.Pow(10, 147455)-1)/(BigInteger.Pow(10, 0)))
+        /// Read here: https://stackoverflow.com/questions/26396856/determine-postgres-numeric-max-min-values       
+        private static readonly BigInteger s_maxValueWithoutScale = BigInteger.Pow(10, MaxPrecision) - 1;
+
+        private static readonly BigInteger s_maxValueWithScale = ((BigInteger.Pow(10, MaxDigitsBeforeDecimalPoint) - 1) / (BigInteger.Pow(10, MaxScale)));
+
+        private static readonly BigInteger s_minValueWithScale = -s_maxValueWithScale;
+
+        private static readonly BigInteger s_minValueWithoutScale = -s_maxValueWithoutScale;
+
+        private static readonly Dictionary<int, BigInteger> s_powersOf10 = new Dictionary<int, BigInteger>();
+
+        // TODO: Don't require a 0 before the decimal point.
+        // TODO: Replace with manual validation if we find this is a performance bottleneck (as it was with BigQuery field name validations see #4976).
+        private static readonly Regex s_validation = new Regex(@"^-?[0-9]+\.?[0-9]*$", RegexOptions.Compiled);
+
+        // Note: the following properties must be declared *after* s_maxValue and s_minValue. Initialization order matters.
+
+        /// <summary>
+        /// Zero represented as a <see cref="PgNumeric"/>. This is the default value for the type.
+        /// </summary>
+        public static PgNumeric Zero { get; } = default;
+
+        /// <summary>
+        /// NaN (Not a number) represented as a <see cref="PgNumeric"/>. 
+        /// </summary>
+        public static PgNumeric NaN { get; } = Parse(NotANumber);
+
+        /// <summary>
+        /// The maximum valid value for <see cref="PgNumeric"/>, equal to BigInteger.Pow(10, MaxPrecision) - 1
+        /// </summary>
+        public static PgNumeric MaxValue { get; } = new PgNumeric(s_maxValueWithoutScale, 0, false);
+
+        /// <summary>
+        /// The minimum valid value for <see cref="PgNumeric"/>, equal to -(BigInteger.Pow(10, MaxPrecision) - 1)
+        /// </summary>
+        public static PgNumeric MinValue { get; } = new PgNumeric(s_minValueWithoutScale, 0, false);
+
+        // The scaled value of PgNumeric would be stored in this variable.
```

Comment:
I think the "would be" in these comments is a bit confusing. That *is* the value.

However, more clarity around what "scaled value" means would definitely be useful. Which direction is this scaling in?

For example, if _scaledValue is 123 and _scale is 1, does that mean the value is 12.3? I suspect so, but just putting that simple example here would really help.